File: Source/Java/Expressive/src/main/java/com/bijington/expressive/functions/string/LengthFunction.java
package com.bijington.expressive.functions.string;

import com.bijington.expressive.exceptions.ExpressiveException;
import com.bijington.expressive.expressions.IExpression;
import com.bijington.expressive.functions.FunctionBase;

/**
 * Created by shaun on 18/07/2016.
 */
public class LengthFunction extends FunctionBase {
    @Override
    public String getName() {
        return "Length";
    }

    @Override
    public Object evaluate(IExpression[] parameters) throws ExpressiveException {
        this.validateParameterCount(parameters, 1, 1);

        Object value = parameters[0].evaluate(this.getVariables());

        if (value == null) return null;

        if (value.getClass().equals(String.class)) {
            return String.class.cast(value).length();
        }
        else {
            return value.toString().length();
        }
    }
}

File: Source/CSharp/Expressive/Expressive/Functions/String/LengthFunction.cs
using Expressive.Expressions;

namespace Expressive.Functions.String
{
    internal class LengthFunction : FunctionBase
    {
        #region FunctionBase Members

        public override string Name
        {
            get
            {
                return "Length";
            }
        }

        public override object Evaluate(IExpression[] parameters, ExpressiveOptions options)
        {
            this.ValidateParameterCount(parameters, 1, 1);

            object value = parameters[0].Evaluate(Variables);

            if (value == null) return null;

            string text = value as string;
            if (text != null)
            {
                return text.Length;
            }
            else
            {
                return value.ToString().Length;
            }
        }

        #endregion
    }
}
